### PR TITLE
PWN-6262 - Off-ramp. Bottom sheet layout 1/2

### PR DESCRIPTION
**Pull request by gslevinkov**
## Jira Ticket

https://p2pvalidator.atlassian.net/browse/PWN-6262

### CODE REVIEW

File: app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockedFragment.kt [hunk 1]
```diff
@@ -26,44 +33,100 @@ class SellLockedFragment :
...
     companion object {
-        fun create(arguments: SellLockedArguments): SellLockedFragment =
-            SellLockedFragment()
-                .withArgs(ARG_SELL_LOCKED to arguments)
+        fun create(details: SellTransactionDetails): SellLockedFragment {
+            require(details.status == SellTransactionStatus.WAITING_FOR_DEPOSIT) {
+                "This fragment is used only if status == waiting for deposit"
+            }
+            return SellLockedFragment()
+                .withArgs(ARG_SELL_LOCKED to details)
```
Comment (1sakov) [inline app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockedFragment.kt:41]: Weird.
Maybe you just create a function only for the determined status then?

**Review by 1sakov — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockedFragment.kt (hunk 1 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockedFragment.kt:41]: it's not a sealed class, i can't pass only determined status there 
i will cast it to sealed class later, in another task, there will be a new field there

File: app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/HomeActionsBottomSheet.kt [hunk 2]
```diff
@@ -55,12 +53,6 @@ class HomeActionsBottomSheet :
         expandToFitAllContent()
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/HomeActionsBottomSheet.kt:53]: remove it from here as lon as it is called in BaseMvpBottomSheet

File: app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt [hunk 3]
```diff
@@ -1,22 +1,197 @@
...
+        when (action) {
+            SellTransactionDetailsButtonAction.SEND -> {
+                val recipient = SearchResult.AddressOnly(AddressState(details.receiverAddress, NetworkType.SOLANA))
+                replaceFragment(NewSendFragment.create(recipient = recipient))
+            }
+            SellTransactionDetailsButtonAction.CLOSE -> {
+                dismissAllowingStateLoss()
+            }
+            SellTransactionDetailsButtonAction.TRY_AGAIN -> {
+                // todo should be a task somewhere to implement try again feature
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt:72]: ticket number if it is exists

File: app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt [hunk 4]
```diff
@@ -1,22 +1,197 @@
...
+            }
+        }
+    }
+
+    private fun setupViews() = with(binding.layoutDetails) {
```
Comment (kabramovp2p) [inline app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt:77]: big big method =(

**Review by kabramovp2p — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/HomeActionsBottomSheet.kt (hunk 2 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/home/ui/main/bottomsheet/HomeActionsBottomSheet.kt:53]: it's not called in BaseMvpBottomSheet

File: app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt (hunk 4 above)
Comment (gslevinkov) [reply app/src/main/java/org/p2p/wallet/moonpay/ui/transaction/SellTransactionDetailsBottomSheet.kt:77]: i know, right? 
i can extract it to class if it helps..

**Review by 1sakov — APPROVED** [review]

File: app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockedFragment.kt (hunk 1 above)
Comment (1sakov) [reply app/src/main/java/org/p2p/wallet/sell/ui/lock/SellLockedFragment.kt:41]: Yeah, would be nice. 
We should avoid throwing exception when creating a fragment

**Pull request merged**